plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt

```diff
@@ -556,15 +554,30 @@ private class DokkaDescriptorVisitor(
         getDocumentation()?.toSourceSetDependent() ?: emptyMap()
 
     private fun ClassDescriptor.resolveClassDescriptionData(): ClassInfo {
-        val interfaces = hashSetOf<DRI>()
-        fun processSuperClasses(supers: List<ClassDescriptor>) {
-            supers.forEach {
-                if(it.kind == ClassKind.INTERFACE) interfaces.add(DRI.from(it))
-                processSuperClasses(it.getSuperInterfaces() + it.getAllSuperclassesWithoutAny())
-            }
+        tailrec fun processSuperClasses(
```

Comment:
I was a little bit confused by this name when starting to read this PR! 
Maybe something like `buildInheritenceTree` would be easier to understand?

Also: Could you explain to me why you are trying to model this inheritance **tree** with a flat data structure like a set?

Reply:
I also think you are creating a lot of information here. From the first reading: You only need the superclasses of inheritance level=0 and all implemented interfaces, right? If this is correct, then we should keep it simple and not do all the extra work.

Reply:
Okey, the name is not the best :)

i am building a whole structure since right now i need inheritance level=0 and all interfaces. In the nearest future i'll need a whole structure to render javadoc's inheritance tree (mby with some small adjustments to cover edge cases). I thought that it will be easier to just do it once and not rework the same piece of code few times

Reply:
I've changed name to `buildInheritanceInformation` since i wouldn't like to suggest that this structure is a tree